app/client/components/user/EditUserForm.tsx

```diff
@@ -0,0 +1,371 @@
+import * as React from "react";
+import { connect } from "react-redux";
+
+import { reduxForm, InjectedFormProps, formValueSelector } from "redux-form";
+
+import { GenderTypes } from "./types";
+import {
+  Wrapper,
+  ModuleBox,
+  ModuleHeader,
+  TopRow,
+  ButtonWrapper,
+  Row,
+  SubRow
+} from "../styledElements";
+import AsyncButton from "./../AsyncButton.jsx";
+import ProfilePicture from "../profilePicture.jsx";
+import GetVerified from "../GetVerified.jsx";
+
+import InputField from "../form/InputField";
+import SelectField from "../form/SelectField";
+import { TransferUsage } from "../../reducers/transferUsage/types";
+import { ReduxState } from "../../reducers/rootReducer";
+import { TransferAccountTypes } from "../transferAccount/types";
+import QrReadingModal from "../qrReadingModal.jsx";
+import { replaceUnderscores } from "../../utils";
+
+export interface IEditUser {
+  firstName?: string;
+  lastName?: string;
+  publicSerialNumber?: string;
+  phone?: string;
+  bio?: string;
+  referredBy?: string;
+  location?: string;
+  businessUsage?: string;
+  usageOtherSpecific?: string;
+  oneTimeCode: number;
+  failedPinAttempts: number;
+  accountType: any[TransferAccountTypes];
+  [key: string]: any;
```

Comment:
What's this needed for/is there a way we can refactor this to not require allowing an index signature?

Reply:
The reason for `[key: string]: any;` is so that custom_attr_keys can be mapped to the form.

`let custom_attr_keys = customAttributes && Object.keys(customAttributes);
    let attr_dict = {};
    custom_attr_keys.map(key => {
      (attr_dict as attr_dict)[key] = customAttributes[key];
      return attr_dict;
    });`

Reply:
We don't know what custom_attributes will be returned. That's why attr_dict and IEditUser need to support.

... although currently the custom_attributes frontend-only supports {key: value} which is 

```
interface attr_dict {
  [key: string]: string;
}
```

Reply:
I could update to that?

Reply:
mmmmmm doesn't really help much. It could make more sense to have a sub-dict that stores custom attributes, and that can be loosely. Means that the system doesn't need to be clever about whats a custom attribute or not either